tests/publiccloud/run_ltp.pm

```diff
@@ -141,18 +157,44 @@ sub run {
 
     assert_script_run($log_start_cmd);
 
-    my $cmd = 'perl -I runltp-ng runltp-ng/runltp-ng ';
-    $cmd .= '--logname=ltp_log --verbose ';
-    $cmd .= '--timeout=1200 ';
-    $cmd .= '--run ' . get_required_var('LTP_COMMAND_FILE') . ' ';
-    $cmd .= '--exclude \'' . get_var('LTP_COMMAND_EXCLUDE') . '\' ' if get_var('LTP_COMMAND_EXCLUDE');
-    $cmd .= '--backend=ssh';
-    $cmd .= ':user=' . $instance->username;
-    $cmd .= ':key_file=' . $instance->ssh_key;
-    $cmd .= ':host=' . $instance->public_ip;
-    $cmd .= ':reset_command=\'' . $reset_cmd . '\'';
-    $cmd .= ':ssh_opts=\'-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no\' ';
-    $cmd .= '--json_filter=openqa ';
+    my $cmd;
+
+    if ($ltp_runtime =~ /^python/i) {
+        zypper_call("in -y python3-paramiko python3-scp");
+
+        my $sut = ':user=' . $instance->username;
+        $sut .= ':sudo=1';
+        $sut .= ':key_file=' . $instance->ssh_key;
+        $sut .= ':host=' . $instance->public_ip;
+        $sut .= ':reset_command=\'' . $reset_cmd . '\'';
+        $sut .= ':hostkey_policy=missing';
+        $sut .= ':known_hosts=/dev/null';
+
+        $cmd = 'python3 runltp-ng/runltp-ng ';
+        $cmd .= "--json-report=$root_dir/result.json ";
+        $cmd .= '--verbose ';
+        $cmd .= '--run-suite ' . get_required_var('LTP_COMMAND_FILE') . ' ';
+        $cmd .= '--skip-tests \'' . get_var('LTP_COMMAND_EXCLUDE') . '\' ' if get_var('LTP_COMMAND_EXCLUDE');
+        $cmd .= '--sut=ssh' . $sut . ' ';
+    } elsif ($ltp_runtime eq 'perl') {
```

Comment:
Let's stay consistent. Either use everywhere the case-insensitive regex matching or nowhere. I don't see why we should use a regex here, it adds mental effort when trying to understand it and the only advantage that we have is that the matching is case insensitive.

Maurizio, please pick your side :slightly_smiling_face: Either everywhere perl regex matching or nowhere in this if ... elsif ... block